elegant: errors in shell command handling

There are two issues with handling of shell commands given in an elegant command file.
See example in `twiss_output.beta_x`: https://www.sirepo.com/elegant#/control/HnD5RV54
`beta_x = "{sddsanalyzebeam whruby.sdds -pipe=out | sdds2stream -pipe -column=betax}"`

- Importing into Sirepo the import of the command and write out seem to be handled just fine, but the input distribution `whruby.sdds` is also being used by `sdds_beam` and Sirepo is mangling the file's name without changing the shell command.

- If using `sirepo.lib.Importer` directly `parse_file fails`, I think because it tries to evaluate the shell command as an RPN command. 

```
/home/vagrant/.pyenv/versions/py3/lib/python3.7/site-packages/sirepo/template/template_common.py:605:subprocess_output ['rpnl', ' {sddsanalyzebeam whruby.sdds -pipe=out | sdds2stream -pipe -column=betax}']: exit=1 err=b'unknown token: {sddsanalyzebeam
unknown token: whruby.sdds
unknown token: -pipe=out
unknown token: |
unknown token: sdds2stream
unknown token: -pipe
unknown token: -column=betax}
'
/home/vagrant/.pyenv/versions/py3/lib/python3.7/site-packages/sirepo/template/elegant.py:210:_model model=command_twiss_output field=beta_x decl=['Beta X', 'RPNValue', '1', 'If matched is zero, the initial values for the X plane.'] value={sddsanalyzebeam whruby.sdds -pipe=out | sdds2stream -pipe -column=betax} exception=expr={sddsanalyzebeam whruby.sdds -pipe=out | sdds2stream -pipe -column=betax} err=invalid
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.7/site-packages/sirepo/lib.py", line 27, in parse_file
    self.__adapter.parse_file(p),
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.7/site-packages/sirepo/template/elegant.py", line 129, in parse_file
    return self._convert(d)
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.7/site-packages/sirepo/template/elegant.py", line 218, in _convert
    _model(m, LatticeUtil.model_name_for_data(m))
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.7/site-packages/sirepo/template/elegant.py", line 195, in _model
    if _rpn('RPNValue', 'Float'):
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.7/site-packages/sirepo/template/elegant.py", line 187, in _rpn
    model[k] = c.eval_var_with_assert(v)
  File "/home/vagrant/.pyenv/versions/py3/lib/python3.7/site-packages/sirepo/template/code_variable.py", line 69, in eval_var_with_assert
    assert not err, f'expr={expr} err={err}'
AssertionError: expr={sddsanalyzebeam whruby.sdds -pipe=out | sdds2stream -pipe -column=betax} err=invalid
```


For the Importer my first thought is that shell commands should be ignored and written out without evaluation. Pure RPN commands should be evaluated still, but this would mean that cases with shell commands inside RPN commands would also be left unevaluated. Though if the command is operating on a file that elegant will generate during the simulation this could still fail since `write_files` does still mangle the output file names.

--- a/sirepo/template/code_variable.py
+++ b/sirepo/template/code_variable.py
@@ -67,7 +67,10 @@ class CodeVar():
     def eval_var_with_assert(self, expr):
         (v, err) = self.eval_var(expr)
         assert not err, f'expr={expr} err={err}'
-        return float(v)
+        try:
+            return float(v)
+        except ValueError:
+            return v
 
     def get_application_data(self, args, schema, ignore_array_values=False):
         from sirepo import simulation_db

--- a/sirepo/template/elegant_lattice_importer.py
+++ b/sirepo/template/elegant_lattice_importer.py
@@ -51,6 +51,9 @@ class ElegantRPNEval(object):
         # sleep                       sleep for number of seconds
         # @                       push command input file
         pkdc('rpn variables={} expr="{}"', var_list, expr)
+        if re.match(r'^\{.+\}$', expr):
+            # It is a shell command
+            return expr, None
         out = elegant_common.subprocess_output(['rpnl', '{} {}'.format(var_list, expr)])
         if out is None:
             return None, 'invalid'

--- a/tests/lib_data/elegant_1/first.ele
+++ b/tests/lib_data/elegant_1/first.ele
@@ -92,7 +92,7 @@
       ! alpha_y = -5.4,
 	!	reference_file = twiss.input,
 	!Twiss Parameters from twiss.input (from S.Shin\..\Method_SShin)
-	beta_x = 4.034896311524342,
+	beta_x = "{sddsanalyzebeam whruby.sdds -pipe=out | sdds2stream -pipe -column=betax}",
 	beta_y = 4.199343696609461,
 	alpha_x = -1.609709872915761,
 	alpha_y = -1.653138844198896

--- a/tests/lib_data/elegant_1/first.ele.out
+++ b/tests/lib_data/elegant_1/first.ele.out
@@ -107,7 +107,7 @@
 &twiss_output
   alpha_x = -1.609709872915761,
   alpha_y = -1.653138844198896,
-  beta_x = 4.034896311524342,
+  beta_x = "({sddsanalyzebeam whruby.sdds -pipe=out | sdds2stream -pipe -column=betax})",
   beta_y = 4.199343696609461,
   filename = "%s.twi",
   matched = 0,

--- a/tests/lib_data/elegant_1/out.json
+++ b/tests/lib_data/elegant_1/out.json
@@ -2931,7 +2931,7 @@
                 "_type": "twiss_output",
                 "alpha_x": -1.609709872915761,
                 "alpha_y": -1.653138844198896,
-                "beta_x": 4.034896311524342,
+                "beta_x": "{sddsanalyzebeam whruby.sdds -pipe=out | sdds2stream -pipe -column=betax}",
                 "beta_y": 4.199343696609461,
                 "cavities_are_drifts_if_matched": "1",
                 "chromatic_tune_spread_half_range": 0.0,

